lock/src/test/java/org/infinispan/lock/impl/lock/RequestExpirationSchedulerTest.java

```diff
@@ -0,0 +1,115 @@
+package org.infinispan.lock.impl.lock;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "clusteredLock.RequestExpirationSchedulerTest")
+public class RequestExpirationSchedulerTest {
+
+   private RequestExpirationScheduler expirationScheduler;
+   private ScheduledExecutorService scheduledExecutorService = mock(ScheduledExecutorService.class);
+   private ScheduledFuture scheduledFutureMock;
+   private CompletableFuture<Boolean> booleanCompletableFuture;
+
+   @BeforeMethod
+   public void createRequestExpirationScheduler() {
+      expirationScheduler = new RequestExpirationScheduler(scheduledExecutorService);
+      scheduledFutureMock = mock(ScheduledFuture.class);
+      booleanCompletableFuture = new CompletableFuture();
+      when(scheduledExecutorService.schedule(any(Runnable.class), anyLong(), any(TimeUnit.class))).thenReturn(scheduledFutureMock);
+   }
+
+   @Test
+   public void testRunScheduling() throws Exception {
+
+      Runnable runnable = expirationScheduler.runScheduling("123", booleanCompletableFuture, 10, TimeUnit.MILLISECONDS);
```

Comment:
I was thinking you could use an argument captor instead. This way you don't have to rewrite your class to have a new method.  You also verify the scheduled executor is called this way. Here is an example: https://github.com/infinispan/infinispan/blob/master/core/src/test/java/org/infinispan/manager/AllClusterExecutorTest.java#L403

Reply:
I really hate them haha we abused in the past with them in other projects xD  but why not for this case

Reply:
Ah well argument captors are great for what they are useful for. And this case is pretty much the case they are designed for.

I am curious as to how you "abused" them because they only have a specific use case normally.